osxcollector/output_filters/virustotal/lookup_urls.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+# -*- coding: utf-8 -*-
+#
+# LookupURLsFilter uses VirusTotal to lookup the URLs in 'LSQuarantineDataURLString' and add 'osxcollector_vturl' key.
+#
+import re
+
+from osxcollector.output_filters.base_filters.output_filter import run_filter
+from osxcollector.output_filters.base_filters. \
+    threat_feed import ThreatFeedFilter
+from osxcollector.output_filters.virustotal.api import VirusTotalApi
+
+
+class LookupURLsFilter(ThreatFeedFilter):
+
+    """A class to find suspicious URLs using VirusTotal API."""
+
+    SCHEMES = re.compile('https?')
+
+    def __init__(self, only_lookup_when=None, is_suspicious_when=None):
+        lookup_when_url_scheme_matches = self._generate_only_lookup_when(only_lookup_when)
+        super(LookupURLsFilter, self).__init__('LSQuarantineDataURLString', 'osxcollector_vturl',
+                                                 only_lookup_when=lookup_when_url_scheme_matches, is_suspicious_when=is_suspicious_when,
+                                                 api_key='virustotal')
+
+    def _generate_only_lookup_when(self, only_lookup_when):
+        """Generates functions that checks whether the blob contains a valid URL
+        in LSQuarantineDataURLString field.
+        """
+        def check_url_scheme(blob):
+            return self.SCHEMES.match(blob.get('LSQuarantineDataURLString')) and (not only_lookup_when or only_lookup_when(blob))
```

Comment:
Very minor:

`blob.get('foo')` will return `None` if `'foo'` is not a key.
In this case since you know `'LSQuarantineDataURLString'` is a key, you could skip using `get()` or add a default param of `''`.